### PR TITLE
Add a basic guide on picking a subject type

### PR DESCRIPTION
**Pull request by josephschorr**

### CODE REVIEW

File: docs/guides/choosing-a-subject-type.md [hunk 1]
```diff
@@ -0,0 +1,89 @@
+# Choosing a Subject Type
```
Comment (jzelinskie) [inline docs/guides/choosing-a-subject-type.md:1]: s/Choosing/Defining

File: docs/guides/choosing-a-subject-type.md [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+
+Within a Permissions System, a [CheckPermission] call is always made between an object representing the *resource* and an object representing a *subject*: The API call returns whether the *subject* has the specified permission on the *resource*.
+
+[CheckPermission]: https://buf.build/authzed/api/docs/main/authzed.api.v1#CheckPermission
+
+The choice of how to represent subjects can have large downstream effects on the design of a permissions system.
```
Comment (jzelinskie) [inline docs/guides/choosing-a-subject-type.md:7]: We probably want to link to the blog post on this somewhere in this section

https://authzed.com/blog/why-model-users/

File: docs/guides/choosing-a-subject-type.md [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+```
+
+As the `sub` field is **guaranteed** to be stable for that particular user (if a compliant OIDC implementation), it is safe to use for checking permissions, as there is no risk that the `sub` will somehow represent a different user later.
+
+:::note
+If you have *multiple* authentication provides, then the recommendation is to define a subject type for *each* provider, to ensure a clean namespace:
+
+```zed
+definition firstprovideruser {}
+definition secondprovideruser {}
```
Comment (jzelinskie) [inline docs/guides/choosing-a-subject-type.md:34]: might be worth using real examples to make this more concrete

File: docs/guides/choosing-a-subject-type.md [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+:::
+
+### Using a primary key
+
+The second most common approach is to have a representation of the subject in another backing data store, typically a relational database.
+
+If such a database exists, and there exists a single row per user, then using the row's primary ID (typically an integer or a UUID) represents another safe ID to use for user:
+
+```
+check resource:someresource view user:1337
```
Comment (jzelinskie) [inline docs/guides/choosing-a-subject-type.md:46]: i think postgres can support reusing autoincre fields, so maybe warn about this?

**Review by jzelinskie — CHANGES_REQUESTED** [review]
looks pretty good, just some additional thoughts

File: docs/guides/choosing-a-subject-type.md (hunk 4 above)
Comment (josephschorr) [reply docs/guides/choosing-a-subject-type.md:46]: Only in certain types, but added a warning

**Comment by josephschorr** [comment]
Updated

**Comment by jzelinskie** [comment]
Maybe we could add two more major concepts to this guide: anonymous users and user hierarchies (admins, teams, managers, users, etc...)

**Comment by josephschorr** [comment]
Updated

**Review by jzelinskie — APPROVED** [review]

**Pull request merged**